src/pycares/__init__.py

```diff
@@ -349,34 +348,33 @@ def _run_safe_shutdown_loop(self) -> None:
             # Block forever until we get a channel to destroy
             channel = self._queue.get()
 
-            # Sleep for 1 second to ensure c-ares has finished processing
-            # Its important that c-ares is past this critcial section
-            # so we use a delay to ensure it has time to finish processing
-            # https://github.com/c-ares/c-ares/blob/4f42928848e8b73d322b15ecbe3e8d753bf8734e/src/lib/ares_process.c#L1422
-            time.sleep(1.0)
+            # Wait for all queries to finish
+            _lib.ares_queue_wait_empty(channel[0], -1)
```

Comment:
If we are here nothing can be holding on to a channel, so we should be able to safely wait for the cancelled queries to end. I don't see any option other than a c-ares internal bug breaking this.